### PR TITLE
Explore Page & Forms

### PR DESCRIPTION
**Pull request by yanevdw**
This PR features the implementation of the explore/search page.

This includes:
- Adding `Explore.tsx` to represent the the explore page.
- Updating the routing configuration for the explore page in `routes/explore.tsx'.
- Adding Formik and Yup to create and validate a form implemented as a searchbar.
- Added additional interfaces to process the API response when searching for comics.
- Added API endpoints in `services/api.ts` to handle searching for comics by genre and by name.

Kindly take note:
- I have selected a few popular genres in the list of genres comics in the API can have, added them to an array and display them as pills that the user can select to search for comics of a certain genre.
- The search bar is mainly to search for a specific comic by name.

### CODE REVIEW

File: src/components/Explore.tsx
Comment (LesediMoko) [inline src/components/Explore.tsx:1]: To shorten the file length and only have code closer to the rendering of the component, maybe separate out the initialisations of those arrays into a different file called defaults or something and just import them in

` import comicGenres, colourCodeGenreLabels, comicGenreIcons from "../default-values.ts" `

File: src/components/Explore.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+            key={comicGenre.toLowerCase()}
+            className={`text-white text-sm rounded-2xl flex flex-row items-center gap-2 ${colourCodeGenreLabels[index]} px-3 py-2`}
+            onClick={() => handleGenreClick(comicGenre)}
+          >
+            {comicGenreIcons[index]}
+            {comicGenre}
+          </label>
+        ))}
+      </div>
+      {selectedGenre && <SearchComicsByGenreComponent genre={selectedGenre} />}
```
Comment (LesediMoko) [inline src/components/Explore.tsx]: Not sure if you need the "Component" suffix on the component name 😅. But if it's to help you recongize it better then that's cool

File: src/services/api.ts
Comment (LesediMoko) [inline src/services/api.ts:1]: Consider having a ` const baseSearchURL = `${baseURL}v1.0/search?` `

The backticks might mess up the formatting of the above code 💀

**Review by LesediMoko — DISMISSED** [review]
Great work, Yané! Just a few notes regarding the formatting of the code but amazing work all-in-all!

File: src/components/Explore.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  const comicGenreIcons = [
+    <PiLightningFill className="text-red-600 text-lg" />,
+    <PiSwordFill className="text-orange-500 text-lg" />,
+    <FaLaugh className="text-yellow-500 text-lg" />,
+    <FaHourglassHalf className="text-lime-500 text-lg" />,
+    <FaWandMagicSparkles className="text-cyan-500 text-lg" />,
+    <GiMagnifyingGlass className="text-purple-400 text-lg" />,
+    <PiAlienFill className="text-green-700 text-lg" />,
+    <IoSunnySharp className="text-pink-500 text-lg" />,
+  ];
```
Comment (Musa-Mabasa) [inline src/components/Explore.tsx]: A suggestion, maybe large mapping data, such as the three above could be in a different file. That way, it doesn't cause clutter in this one component file.

File: src/components/Explore.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+            ></img>
+            <div className="image-overlay h-full w-full rounded-lg bg-gradient-to-t from-plum absolute flex items-end justify-center text-center p-0 m-0">
+              <p className="p-2 font-sans text-white">{searchResult.title}</p>
+            </div>
+          </div>
+        ))}
+      </div>
+    </>
+  );
+}
```
Comment (Musa-Mabasa) [inline src/components/Explore.tsx]: Each component here should be placed in it's own TSX file.

File: src/components/Explore.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+      <h2 className="font-semibold text-white text-lg">Search Results</h2>
+      <div className="search-container w-full h-3/10 carousel carousel-center space-x-4 bg-neutral rounded-lg bg-transparent overflow-y-hidden flex align-center py-4 mb-4">
+        {filteredComics?.map((searchResult: SearchComicContent) => (
+          <div
+            key={searchResult.slug}
+            className="carousel-item w-2/6 h-full p-0 m-0 flex flex-col justify-center relative first:pl-0 hover:cursor-pointer md:w-1/10 lg:w-1/10"
+            onClick={() =>
+              (
+                document.getElementById(
+                  `search-comic-popup-${searchResult.slug}`
```
Comment (Musa-Mabasa) [inline src/components/Explore.tsx]: Safety checks on `searchResult` and it's nested properties. Same with everything below ⬇️ and above ⬆️

File: src/components/Explore.tsx
Comment (yanevdw) [reply src/components/Explore.tsx:1]: Thank you, I'll do this.

File: src/components/Explore.tsx (hunk 1 above)
Comment (yanevdw) [reply src/components/Explore.tsx]: Removed them 😄

File: src/services/api.ts
Comment (yanevdw) [reply src/services/api.ts:1]: Done ✨

File: src/components/Home.tsx [hunk 5]
```diff
@@ -92,9 +92,6 @@ function TopComicComponent({ comicType }: { comicType: string }) {
             <div className="image-overlay h-full w-full rounded-lg bg-gradient-to-t from-plum absolute flex items-end justify-center text-center p-0 m-0">
               <p className="p-2 font-sans text-white">{comicRec.title}</p>
```
Comment (Musa-Mabasa) [inline src/components/Home.tsx]: Safety check here. I also recommend using nullish coalescing: `comicRec?.title ?? "PlaceHolderTitle"` to show a place holder title of some sorts if title is undefined or null. That's just a suggestion, you don't have to do it like that if it's not to your liking.

**Review by Musa-Mabasa — CHANGES_REQUESTED** [review]
A very nice use of forms!  Some safety checks and code refactoring from my comments. Overall, well done! 👍

File: src/components/Home.tsx (hunk 5 above)
Comment (yanevdw) [reply src/components/Home.tsx]: Thank you. I will add safety checks and come back to nullish coalescing if I have time in the future.

**Review by LesediMoko — APPROVED** [review]
All comments have been attended to. Great work, Yané!

**Review by Musa-Mabasa — APPROVED** [review]
Everything looks good! 👍

**Pull request merged**